internal/network/manifests_generator.go

```diff
@@ -68,6 +69,81 @@ spec:
   osImageURL: ""
 `
 
+const snoDnsmasqConf = `
+address=/{{.CLUSTER_NAME}}.{{.DNS_DOMAIN}}/{{.HOST_IP}}
+addn-hosts=/etc/api-int.host
+`
+
+const apiIntHosts = `
+{{.HOST_IP}} api-int api-int.{{.CLUSTER_NAME}}.{{.DNS_DOMAIN}}
+`
+
+const forceDnsDispatcherScript = `
+export IP="{{.HOST_IP}}"
+if [ "$2" = "dhcp4-change" ] || [ "$2" = "dhcp6-change" ] || [ "$2" = "up" ] || [ "$2" = "connectivity-change" ]; then
+    grep -q $IP /etc/resolv.conf
+    if [ "$?" != "0" ] ; then
+      sed -i "s/{{.CLUSTER_NAME}}.{{.DNS_DOMAIN}}//" /etc/resolv.conf
+      sed -i "s/search /search {{.CLUSTER_NAME}}.{{.DNS_DOMAIN}} /" /etc/resolv.conf
+      sed -i "0,/nameserver/s/nameserver/nameserver $IP\nnameserver/" /etc/resolv.conf
+    fi
+fi
+`
+
+const dnsMachineConfigManifest = `
+apiVersion: machineconfiguration.openshift.io/v1
+kind: MachineConfig
+metadata:
+  labels:
+    machineconfiguration.openshift.io/role: {{.ROLE}}
+  name: 99-{{.ROLE}}s-dnsmasq-configuration
+spec:
+  config:
+    ignition:
+      config: {}
+      security:
+        tls: {}
+      timeouts: {}
+      version: 2.2.0
+    networkd: {}
+    passwd: {}
+    storage:
+      files:
+      - contents:
+          source: data:text/plain;charset=utf-8;base64,{{.DNSMASQ_CONF}}
+          verification: {}
+        filesystem: root
+        mode: 420
+        path: /etc/dnsmasq.d/sno.conf
+      - contents:
+          source: data:text/plain;charset=utf-8;base64,{{.API_INT_HOST_FILE}}
+          verification: {}
+        filesystem: root
+        mode: 420
+        path: /etc/api-int.host
+      - contents:
+          source: data:text/plain;charset=utf-8;base64,{{.FORCE_DNS_SCRIPT}}
+          verification: {}
+        filesystem: root
+        mode: 365
+        path: /etc/NetworkManager/dispatcher.d/forcedns
+    systemd:
+      units:
+      - name: dnsmasq.service
+        enabled: true
+        contents: |
+         [Unit]
+         Description=Run dnsmasq to provide local dns
+         Before=kubelet.service crio.service
+         After=network.target
+         
```

Comment:
Dont we need this:
https://github.com/openshift/machine-config-operator/pull/2374/files#diff-a7463093825916371a20c826e431e678ab68f62cce764c128bab5c292f9292a5R8

Reply:
Or it's mostly for BM with VIPs.